### PR TITLE
Adds an APC sanity unit test

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Adds an APC sanity unit test
## Why It's Good For The Game

Areas that need APCs should have APCs and areas that don't shouldn't have APCs. No area should have multiple APCs.
## Changelog
:cl:
add: Adds an APC sanity unit test
/:cl:

### CODE REVIEW

**Comment by XElectricX** [comment]
This is beyond overdue. Thanks Ivan.

**Comment by ivanmixo** [comment]
It's kinda not ideal because it only tests the map currently being ran, should figure out a way to somehow make it so runs on all maps, although this would inflate the hell out of the actions. TG does it though. But we have a lot more maps than them. Bleh.

File: code/modules/unit_tests/apc_area_sanity.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/datum/unit_test/area_apc_sanity/Run()
+	for(var/area/tested_area in world)
+		var/list/cached_apc_list = tested_area.get_apc_list()
+		if(!length(cached_apc_list) && tested_area.requires_power && !tested_area.always_unpowered)
+			Fail("[tested_area.type] doesn't have an APC while it needs one.")
+		if(length(cached_apc_list) && (!tested_area.requires_power || tested_area.always_unpowered))
+			Fail("[tested_area] has APC(s) while not needing power.")
+			for(var/obj/machinery/power/apc/found_apc in cached_apc_list)
+				Fail("Unneded APC at [found_apc.x], [found_apc.y], [found_apc.z]")
```
Comment (QualityVan) [inline code/modules/unit_tests/apc_area_sanity.dm:11]:
```suggestion
				Fail("Unneeded APC at [found_apc.x], [found_apc.y], [found_apc.z]")
```

**Review by RipGrayson — APPROVED** [review]

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Also grayson changed file codes again or whatever theyre called

File: code/modules/unit_tests/apc_area_sanity.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/datum/unit_test/area_apc_sanity
+
+/datum/unit_test/area_apc_sanity/Run()
+	for(var/area/tested_area in world)
```
[the remark this answers is not in the document]
Comment (ivanmixo) [reply code/modules/unit_tests/apc_area_sanity.dm:4]: I guess I could restrict it to just ground and station.

**Pull request merged**